plonky2x/core/src/backend/function/mod.rs

```diff
@@ -81,36 +89,32 @@ impl<C: Circuit> Plonky2xFunction for C {
             let contract_path = format!("{}/FunctionVerifier.sol", args.build_dir);
             let mut contract_file = File::create(&contract_path).unwrap();
 
-            let circuit_digest_bytes = circuit
+            info!("First building wrapper circuit to get the wrapper circuit digest...");
+            // The wrapper circuit digest will get saved in the Solidity smart contract,
+            // which will use this value as a public input `VerifierDigest` in the Gnark plonky2 verifier
+            let wrapped_circuit = WrappedCircuit::<L, WrapperParameters, D>::build(circuit);
+
+            let mut circuit_digest_bytes = wrapped_circuit
+                .wrapper_circuit
                 .data
                 .verifier_only
                 .circuit_digest
-                .to_vec()
-                .iter()
-                .flat_map(|e| e.to_canonical_u64().to_be_bytes())
-                .collect::<Vec<u8>>();
-            let full_circuit_digest_bytes = circuit
-                .data
-                .verifier_only
-                .constants_sigmas_cap
-                .0
-                .iter()
-                .flat_map(|x| {
-                    x.elements
-                        .iter()
-                        .flat_map(|e| e.to_canonical_u64().to_be_bytes())
-                })
-                .chain(circuit_digest_bytes.iter().copied())
-                .collect::<Vec<u8>>();
-            let circuit_digest_hash = sha2::Sha256::digest(full_circuit_digest_bytes);
-            assert!(
-                circuit_digest_hash.len() <= 32,
-                "circuit digest must be <= 32 bytes"
-            );
-
+                .to_bytes();
+            // to_bytes() returns the representation as LE, but we want to save it on-chain
+            // as BE because that is the format of the public input to the Gnark plonky2 verifier.
+            circuit_digest_bytes.reverse();
+
+            // The VerifierDigest is stored on-chain as a bytes32, so we need to pad it with 0s
+            // to store it in the Solidity smart contract.
+            // Note that we don't need to do any sort of truncation of the top bits because the
+            // circuit digest already lives in the bn254 field because the WrappedCircuit config
```

Comment:
nit: why is bn254 lowercase but next line it's capitalized

Reply:
nit: if you want line breaks by sentence but with no extra space are unusual in this repo

either collapse to 100 cols or add another line between line breaks